### PR TITLE
Permit tests that fail certain assertions to run in !DEBUG

### PR DESCRIPTION
**Pull request by gafter**
See #25702 for details.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs [hunk 1]
```diff
@@ -849,7 +849,8 @@ public void IntoScriptBlock()
                 Diagnostic(ErrorCode.WRN_UnreferencedLabel, "L1").WithLocation(6, 5));
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/25702")]
+#if !DEBUG // https://github.com/dotnet/roslyn/issues/25702
+        [Fact]
```
[the remark this answers is not in the document]
Comment (gafter) [reply src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs]: Thanks, I looked but didn't find that.  I will switch it.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/WinMdTests.cs [hunk 2]
```diff
@@ -291,6 +290,7 @@ .maxstack  2
 }");
             });
         }
+#endif
```
Comment (AlekseyTs) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/WinMdTests.cs]: It doesn't look like there is a matching ```#if``` directive.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/WinMdTests.cs (hunk 2 above)
Comment (gafter) [reply src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/WinMdTests.cs]: Yes, this PR isn't finished, which is why it marked for personal review only.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a review of this test-only change?

**Pull request merged**